src/main/java/be/seeseemelk/mockbukkit/MockBukkit.java

```diff
@@ -267,6 +269,117 @@ public static void loadJar(@NotNull File jarFile) throws InvalidPluginException
 	}
 
+	/**
+	 * Loads and enables the Plugin with a specified Config File. It receives the {@code config.yml} as a InputStream
+	 * to load.
+	 *
+	 * @param <T>        The plugin's main class to load.
+	 * @param plugin     The plugin to load for mocking.
+	 * @param configStream The {@code InputStream} of the {@code config.yml} file to load.
+	 * @param parameters Extra parameters to pass on to the plugin constructor.
+	 * @return An instance of the plugin's main class.
+	 * @apiNote The File name can't be the same as the one the plugin loads by default
+	 */
+	public static <T extends JavaPlugin> @NotNull T loadWithConfig(
+			@NotNull Class<T> plugin,
+			InputStream configStream,
+			Object... parameters
+	)
+	{
+		FileConfiguration config = new YamlConfiguration();
+		try
+		{
+			config.load(new InputStreamReader(configStream));
+		}
+		catch (IOException | InvalidConfigurationException e)
+		{
+			throw new RuntimeException("Couldn't read config input stream",e);
+		}
+		return loadWithConfig(plugin, config, parameters);
+	}
+
+
+	/**
+	 * Loads and enables the Plugin with a specified Config File. It receives the {@code config.yml} to use as File
+	 * to load.
+	 *
+	 * @param <T>        The plugin's main class to load.
+	 * @param plugin     The plugin to load for mocking.
+	 * @param configFile The File of the {@code config.yml} file to load.
+	 * @param parameters Extra parameters to pass on to the plugin constructor.
+	 * @return An instance of the plugin's main class.
+	 * @apiNote The File name can't be the same as the one the plugin loads by default
+	 */
+	public static <T extends JavaPlugin> @NotNull T loadWithConfig(
+			@NotNull Class<T> plugin,
+			File configFile,
+			Object... parameters
+	)
+	{
+		FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+		return loadWithConfig(plugin, config, parameters);
+	}
+
+	/**
+	 * Loads and enables the Plugin with a specified Config. It receives the {@code FileConfiguration} to load.
+	 *
+	 * @param <T>               The plugin's main class to load.
+	 * @param plugin            The plugin to load for mocking.
+	 * @param fileConfiguration The {@code FileConfiguration} to load.
+	 * @param parameters        Extra parameters to pass on to the plugin constructor.
+	 * @return An instance of the plugin's main class.
+	 * @apiNote The File name can't be the same as the one the plugin loads by default
+	 */
+	public static <T extends JavaPlugin> @NotNull T loadWithConfig(
+			@NotNull Class<T> plugin,
+			FileConfiguration fileConfiguration,
+			Object... parameters
+	)
+	{
+		ensureMocking();
+		PluginManagerMock pluginManager = mock.getPluginManager();
+		JavaPlugin instance = pluginManager.loadPlugin(plugin, parameters);
+
+		try
+		{
+			if (!instance.
+					getClass()
+					.getSuperclass()
+					.getSuperclass()
+					.getName()
+					.equals(BUKKIT_PLUGIN_SUPERCLASS))
+			{
+				throw new IllegalStateException("Invalid plugin superclass");
+			}
```

Comment:
Is there not an edgecase scenario where the plugin extends another plugin. Who knows right? This explicitly requires the class `org.bukkit.plugin.java.JavaPlugin` to be what the plugin extends.

i would remove this whole section.

Reply:
Thei issue might be that we can't get the field if JavaPlugin is not the superclass of the superclass

Reply:
It'll work regardless of how far up the inheritance chain it is, right? Can we recursively search through superclasses until we find it and only fail if it isn't anywhere in the chain?